File: carRent/src/main/java/com/carRent/carRent/services/customer/CustomerService.java
package com.carRent.carRent.services.customer;

import java.util.List;

import com.carRent.carRent.dto.BookCarDto;
import com.carRent.carRent.dto.CarDto;
import com.carRent.carRent.dto.CarDtoListDto;
import com.carRent.carRent.dto.SearchCarDto;
import com.carRent.carRent.entity.User;

public interface CustomerService {
   List<CarDto> getAllCars();

   boolean bookCar(BookCarDto bookCarDto);

   CarDto getCarById(Long id);

   List<BookCarDto> getBookingsById(Long userId);

   CarDtoListDto searchCar(SearchCarDto searchCarDto);

   public boolean cancelBooking(Long bookingId,String status) ;

   User updateUser(Long id, User updatedUser);

}

File: carRentFront/src/app/modules/customer/service/customer.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { StorageService } from '../../../auth/services/storage/storage.service';
const URL = 'http://localhost:8080';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  constructor(private http: HttpClient, private storageService: StorageService) { }


  createAuthorizationHeader(): HttpHeaders {
    let authHeaders = new HttpHeaders();
    return authHeaders.set('Authorization', 'Bearer ' + this.storageService.getToken());
  }

  getAllCars() {
    const headers = this.createAuthorizationHeader();
    return this.http.get(URL + '/api/customer/cars', { headers });
  }

  getCarById(id: any) {
    const headers = this.createAuthorizationHeader();
    return this.http.get(URL + '/api/customer/cars/' + id, { headers });
  }

  bookCar(bookCarDto: any) {
    const headers = this.createAuthorizationHeader();
    return this.http.post(URL + '/api/customer/cars/book', bookCarDto, { headers });
  }

  getBookingsById() {
    const headers = this.createAuthorizationHeader();
    return this.http.get(URL + '/api/customer/cars/bookings/' + this.storageService.getUser().id, { headers });
  }
  searchCar(searchCarDto:any){
    const headers = this.createAuthorizationHeader();
    return this.http.post(URL + '/api/customer/cars/search', searchCarDto, { headers });
  }

  cancelBooking(bookingId: any,status:any) {
    const headers = this.createAuthorizationHeader();
    return this.http.get(URL + '/api/customer/cars/cancel/' + bookingId+"/"+status, { headers });
  }

  updateUser(id: any, userDto: any) {
    const headers = this.createAuthorizationHeader();
    return this.http.put(URL + '/api/customer/' + id, userDto, { headers });
  }
}
